hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +48,238 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
     // sync the embedded group
     m_cl->setEmbeddedGroup(m_embed_group);
 
+    // check for collision warnings
+    checkCollisionWarnings(timestep);
+
     // set random grid shift
     m_cl->drawGridShift(timestep);
 
     // update cell list
     m_cl->compute(timestep);
 
+    // create auxillary array for rigid bodies
+    if (m_embed_group)
+        {
+        // resize initial velocity array
+        unsigned int N_tot = m_embed_group->getNumMembers();
+        if (N_tot > m_initial_velocity.getNumElements())
+            {
+            GPUArray<Scalar4> initial_velocity(N_tot, m_exec_conf);
+            m_initial_velocity.swap(initial_velocity);
+            }
+        unsigned int Np_tot = m_pdata->getN();
+        if (Np_tot > m_linmom_change.getNumElements())
+            {
+            GPUArray<Scalar3> linmom_change(Np_tot, m_exec_conf);
+            m_linmom_change.swap(linmom_change);
+            GPUArray<Scalar4> angmom_change(Np_tot, m_exec_conf);
+            m_angmom_change.swap(angmom_change);
+            }
```

Comment:
Try to use more descriptive variable names if they might be confused. Also, please remember to use `const`:
```suggestion
        const unsigned int num_group = m_embed_group->getNumMembers();
        if (num_group > m_initial_velocity.getNumElements())
            {
            GPUArray<Scalar4> initial_velocity(num_group, m_exec_conf);
            m_initial_velocity.swap(initial_velocity);
            }

        const unsigned int num_total = m_pdata->getN();
        if (num_total > m_linmom_change.getNumElements())
            {
            GPUArray<Scalar3> linmom_change(num_total, m_exec_conf);
            m_linmom_change.swap(linmom_change);
            GPUArray<Scalar4> angmom_change(num_total, m_exec_conf);
            m_angmom_change.swap(angmom_change);
            }
```